CMakeLists.txt

```diff
@@ -472,6 +470,31 @@ if (BUILD_TESTING)
     file (GLOB TEST_LD_PRELOAD "tests/LD_PRELOAD/*.c")
     add_library(allocator_overrides SHARED ${TEST_LD_PRELOAD})
 
+    set(VALGRIND_DEFAULT " \
+    --leak-check=full \
+    --leak-resolution=high \
+    --trace-children=yes \
+    --run-libc-freeres=no \
+    -q --error-exitcode=123 \
+    --error-limit=no \
+    --num-callers=40 \
+    --undef-value-errors=no \
+    --suppressions=valgrind.suppressions")
+
+    set(VALGRIND_PENDANTIC "${VALGRIND_DEFAULT} --errors-for-leak-kinds=all")
+
+    # Add pendentic Valgrind check for Libcrypto is openssl-1.1.1.
+    # Default Valgrind tests ignores "Still reachable" leak and we want to enable pedantic
+    # valgrind check incrementally, so we will do pedantic check for Libcrypto=openssl-1.1.1 for now.
+    # Tracking issue: https://github.com/aws/s2n-tls/issues/3758
+    if ($ENV{S2N_LIBCRYPTO} MATCHES "openssl-1.1.1")
+        set(MEMORYCHECK_COMMAND_OPTIONS ${VALGRIND_PENDANTIC})
+    else()
+        set(MEMORYCHECK_COMMAND_OPTIONS ${VALGRIND_DEFAULT})
+    endif()
+
+    set(CTEST_MEMORYCHECK_TYPE "Valgrind")
```

Comment:
Let's explicitly call out that these CTEST configuration variables need to be set before `include(CTest)`  is called.